flips/20211007-transaction-fees.md

```diff
@@ -0,0 +1,327 @@
+
+From the users perspective the implementation of transaction fees should satisfy the following criteria:
+
+- Fees should be easy to understand:
+    - It should be made clear why the transaction fees exist in the form they do.
+    - It should be easy to explain how the fees are calculated.
+- It should be possible to calculate the minimum and maximum boundaries of the transaction fees for a transaction (within a reasonable margin), before actually sending a transaction.
+- The price of sending a specific transaction should not fluctuate over time so quickly that the user cannot respond to the fluctuation.
+- It should be possible to see the details of the transaction fee calculation after the transaction is executed.
+
+## Proposed design
+
+The idea behind this design is to break down the transaction fees into smaller parts that are easier to define, control, and implement.
+
+The motivation behind the first separation is to separate the fees into a part that can be known before executing the transaction script, inclusion fees, and a part that can only be know by executing the transaction script, execution fees. 
+
+The inclusion part of fees accounts for the resources needed by a transaction because of the transaction's properties (that can be known without executing the transaction's script). Transaction properties are things like the byte size of the transaction and the number of signatures. Some examples of resources that will be needed are the bandwidth to transfer the transaction from node to node, the verification of signatures and the verification of the sequence number.
+
+
+The execution part of fees accounts for the approximate operational cost of executing the transaction script, passing the results to the verification node and verifying the results. Execution fees cannot be know without actually executing the transaction script. In order to avoid having unexpectedly high execution fees, each transaction should specify a stopping limit for the execution fees.
+
+The inclusion fees and the execution fees can be further split into effort and an the cost of that effort in units of FLOW. The effort required for a specific transaction should be independent of when the transaction is sent[^1] and should only depend on the properties of the transaction. The cost of the effort required to execute a transaction can be a factor with the unit of FLOW per effort multiplied by the effort, however in general terms the price of effort should be a function of effort itself. This would allow increasing the price of effort if a lot of effort was used in a transaction. 
```

Comment:
```suggestion
The inclusion fees and the execution fees can be further split into effort and the cost of that effort in units of FLOW. The effort required for a specific transaction should be independent of when the transaction is sent[^1] and should only depend on the properties of the transaction.
The cost of the effort required to execute a transaction can be a factor with the unit of FLOW per effort multiplied by the effort, however in general terms the price of effort should be a function of effort itself. This would allow increasing the price of effort if a lot of effort was used in a transaction. 
```